Bug: cli-proxy image not included in download_docker_images.sh pre-pull

## Summary

When a workflow uses `features: cli-proxy: true`, the compiler generates a lock file that:
1. Sets `--skip-pull` on the `awf` command (disabling Docker image pulls at runtime)
2. Pre-pulls images via `download_docker_images.sh` — but only `agent`, `api-proxy`, and `squid`
3. Does **not** include `ghcr.io/github/gh-aw-firewall/cli-proxy:<version>` in the pre-pull list

At runtime, AWF tries to create the `awf-cli-proxy` container, fails because the image was never pulled, and `--skip-pull` prevents pulling it on demand.

## Error

```
Container awf-cli-proxy  Creating
Container awf-cli-proxy  Error response from daemon: No such image: ghcr.io/github/gh-aw-firewall/cli-proxy:0.25.18
Error response from daemon: No such image: ghcr.io/github/gh-aw-firewall/cli-proxy:0.25.18
[ERROR] Failed to start containers: Error: Command failed with exit code 1: docker compose up -d --pull never
```

## Reproduction

Any workflow with `features: cli-proxy: true` compiled with gh-aw v0.67.4:

```yaml
features:
  cli-proxy: true
```

The compiled lock file will have a `download_docker_images.sh` step like:

```bash
bash "${RUNNER_TEMP}/gh-aw/actions/download_docker_images.sh" \
  ghcr.io/github/gh-aw-firewall/agent:0.25.18 \
  ghcr.io/github/gh-aw-firewall/api-proxy:0.25.18 \
  ghcr.io/github/gh-aw-firewall/squid:0.25.18 \
  ghcr.io/github/gh-aw-mcpg:v0.2.17 \
  ...
```

Note: `cli-proxy` is missing from this list.

## Expected Behavior

When `features: cli-proxy: true` is set, the compiler should add `ghcr.io/github/gh-aw-firewall/cli-proxy:<version>` to the `download_docker_images.sh` pre-pull list so the image is available when AWF starts with `--skip-pull`.

## Impact

This affects **any repo** using `features: cli-proxy: true` — not just gh-aw-firewall. The cli-proxy image IS published to GHCR as part of the release process (`release.yml` `build-cli-proxy` job), so the fix is just to add it to the pre-pull step.

## Workaround

In gh-aw-firewall we work around this via our post-processing script which replaces `--image-tag/--skip-pull` with `--build-local` (builds all containers from source). Other repos have no workaround.

## Affected Version

gh-aw v0.67.4, AWF v0.25.18

## Related

- Failed run: https://github.com/github/gh-aw-firewall/actions/runs/24215613588
- Failed run: https://github.com/github/gh-aw-firewall/actions/runs/24219039661

--- a/pkg/workflow/docker.go
+++ b/pkg/workflow/docker.go
@@ -99,6 +99,18 @@ func collectDockerImages(tools map[string]any, workflowData *WorkflowData, actio
 				dockerLog.Printf("Added AWF api-proxy sidecar container: %s", apiProxyImage)
 			}
 		}
+
+		// Add cli-proxy sidecar container when the cli-proxy feature flag is enabled
+		// and the AWF version supports it. Without this, --skip-pull causes AWF to fail
+		// because the cli-proxy image was never pulled.
+		if isFeatureEnabled(constants.CliProxyFeatureFlag, workflowData) && awfSupportsCliProxy(firewallConfig) {
+			cliProxyImage := constants.DefaultFirewallRegistry + "/cli-proxy:" + awfImageTag
+			if !imageSet[cliProxyImage] {
+				images = append(images, cliProxyImage)
+				imageSet[cliProxyImage] = true
+				dockerLog.Printf("Added AWF cli-proxy sidecar container: %s", cliProxyImage)
+			}
+		}
 	}
 
 	// Collect sandbox.mcp container (MCP gateway)

--- a/pkg/workflow/docker_cli_proxy_test.go
+++ b/pkg/workflow/docker_cli_proxy_test.go
@@ -1,0 +1,93 @@
+package workflow
+
+import (
+	"slices"
+	"testing"
+
+	"github.com/github/gh-aw/pkg/constants"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+)
+
+func TestCollectDockerImages_CliProxy(t *testing.T) {
+	// Use a version without "v" prefix — getAWFImageTag strips it
+	awfImageTag := "0.25.18"
+
+	t.Run("includes cli-proxy image when feature flag is enabled", func(t *testing.T) {
+		workflowData := &WorkflowData{
+			AI: "claude",
+			NetworkPermissions: &NetworkPermissions{
+				Firewall: &FirewallConfig{
+					Enabled: true,
+					Version: awfImageTag,
+				},
+			},
+			Features: map[string]any{"cli-proxy": true},
+		}
+
+		images := collectDockerImages(nil, workflowData, ActionModeRelease)
+
+		cliProxyImage := constants.DefaultFirewallRegistry + "/cli-proxy:" + awfImageTag
+		assert.True(t, slices.Contains(images, cliProxyImage),
+			"Expected cli-proxy image %q in collected images, got: %v", cliProxyImage, images)
+	})
+
+	t.Run("excludes cli-proxy image when feature flag is absent", func(t *testing.T) {
+		workflowData := &WorkflowData{
+			AI: "claude",
+			NetworkPermissions: &NetworkPermissions{
+				Firewall: &FirewallConfig{
+					Enabled: true,
+					Version: awfImageTag,
+				},
+			},
+		}
+
+		images := collectDockerImages(nil, workflowData, ActionModeRelease)
+
+		cliProxyImage := constants.DefaultFirewallRegistry + "/cli-proxy:" + awfImageTag
+		assert.False(t, slices.Contains(images, cliProxyImage),
+			"Did not expect cli-proxy image %q in collected images without feature flag, got: %v", cliProxyImage, images)
+	})
+
+	t.Run("excludes cli-proxy image when AWF version is too old", func(t *testing.T) {
+		workflowData := &WorkflowData{
+			AI: "claude",
+			NetworkPermissions: &NetworkPermissions{
+				Firewall: &FirewallConfig{
+					Enabled: true,
+					Version: "v0.25.16", // older than AWFCliProxyMinVersion
+				},
+			},
+			Features: map[string]any{"cli-proxy": true},
+		}
+
+		images := collectDockerImages(nil, workflowData, ActionModeRelease)
+
+		// Should not include cli-proxy for an old AWF version
+		for _, img := range images {
+			assert.NotContains(t, img, "/cli-proxy:",
+				"Should not include cli-proxy image when AWF version is too old")
+		}
+	})
+
+	t.Run("cli-proxy image uses correct AWF image tag", func(t *testing.T) {
+		customTag := "0.26.0"
+		workflowData := &WorkflowData{
+			AI: "copilot",
+			NetworkPermissions: &NetworkPermissions{
+				Firewall: &FirewallConfig{
+					Enabled: true,
+					Version: customTag,
+				},
+			},
+			Features: map[string]any{"cli-proxy": true},
+		}
+
+		images := collectDockerImages(nil, workflowData, ActionModeRelease)
+
+		expectedImage := constants.DefaultFirewallRegistry + "/cli-proxy:" + customTag
+		require.True(t, slices.Contains(images, expectedImage),
+			"Expected cli-proxy image %q with custom tag, got: %v", expectedImage, images)
+	})
+}

